src/slider/SliderTrack.ts

```diff
@@ -4,35 +4,37 @@
  * We improved the hook [useSlider](https://github.com/chakra-ui/chakra-ui/blob/af613020125265914a9dcb74c92a07a16aa4ff8e/packages/slider/src/use-slider.ts)
  * to work with Reakit System
  */
-import { createComponent, createHook } from "reakit-system";
-import { BoxHTMLProps, useBox } from "reakit";
 import { useId } from "@chakra-ui/hooks";
+import { BoxHTMLProps, useBox } from "reakit";
 import { mergeRefs, dataAttr } from "@chakra-ui/utils";
+import { createComponent, createHook } from "reakit-system";
 
-import { UseSliderReturn } from "./SliderState";
 import { SLIDER_TRACK_KEYS } from "./__keys";
+import { UseSliderReturn } from "./SliderState";
 
 export type useSliderTrackOptions = UseSliderReturn & {
   /**
-   * The base `id` to use for the sliderTrack and it's components
+   * The base `id` to use for the sliderTrack
    */
   id?: string;
 };
```

Comment:
I guess we should pick only the types which the component is using. 

```ts
export type useSliderTrackOptions = Pick<
  UseSliderReturn,
  "refs" | "state" | "styles"
> & {
  /**
   * The base `id` to use for the sliderTrack
   */
  id?: string;
};
```

Reply:
Same with all other components.

Reply:
What is the advantage or use in picking the variables only we use in a particular component.

I just dont know, can you help me understand.

I think it as since we are spreading all the values from state, I just don't pick it.

Reply:
Yes we don't see any issues because we are spreading the state directly to the components, but i think it's better to be explicit about types it would help end users understand which component takes which props (even tho they would not care about it) & also would help us reduce confusion in the long run.

Yeah i know picking the types is a bit redundant and a bit of a chore but imo it's best we pick them.

For example:

the <Slider /> does not take the action, handler props but if we directly set the `UseSliderReturn` those props also shows as required props. 

Without Pick:
![slider-props-1](https://user-images.githubusercontent.com/35374649/92078610-31a97a80-eddc-11ea-8d3b-9ab978ca6725.jpg)

With Pick:
![slider-props-2](https://user-images.githubusercontent.com/35374649/92078626-366e2e80-eddc-11ea-96a9-1ee678081022.jpg)


Do this on your free time, it's not high priority, or let me know if can also fix them if you want. :D

Reply:
No Worries, I just learned this, I will do it myself, Thanks much for deeps explanation 🥇

Reply:
Awesome!